From 69285365e16e2bfd77aecfadf2cbaad84ea319dc Mon Sep 17 00:00:00 2001
From: M0n7y5 <17201053+M0n7y5@users.noreply.github.com>
Date: Tue, 11 Aug 2026 21:55:22 +0200
Subject: [PATCH 145/162] mmdevapi: Reject wave formats with zero samples per
 seconds.

That currently triggers an assertion the PulseAudio library.

Wine-Bug: not filed
Upstream: f8470906c2540d184a65989c8f0722b9da3d852a
Author: Giovanni Mascellani <gmascellani@codeweavers.com>
---
 dlls/mmdevapi/client.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/mmdevapi/client.c b/dlls/mmdevapi/client.c
index e0d137f823d..6d86bffe440 100644
--- a/dlls/mmdevapi/client.c
+++ b/dlls/mmdevapi/client.c
@@ -337,6 +337,9 @@ static HRESULT validate_wfx(const WAVEFORMATEX *fmt, AUDCLNT_SHAREMODE share_mod
     if (FAILED(ret))
         return ret;
 
+    if (fmt->nSamplesPerSec == 0)
+        ret = E_INVALIDARG;
+
     switch (fmt->wFormatTag) {
         case WAVE_FORMAT_EXTENSIBLE:
             if ((fmt->cbSize != sizeof(WAVEFORMATEXTENSIBLE) - sizeof(WAVEFORMATEX) &&
-- 
2.55.0

